cargo-frc/src/toolchain.rs

```diff
@@ -0,0 +1,160 @@
+#[cfg(target_os = "linux")]
+mod os {
+    use std::path::PathBuf;
+
+    pub fn install_dir() -> PathBuf {
+        dirs::home_dir()
+            .expect("Can't determine home directory")
+            .join("wpilib")
+        //.ok_or("Can't determine home directory".to_owned())
+    }
+
+    pub const TOOLCHAIN_URL_2019: &str = "https://github.com/wpilibsuite/toolchain-builder/releases/download/v2019-3/FRC-2019-Linux-Toolchain-6.3.0.tar.gz";
+    pub const TOOLCHAIN_URL_2020: &str = "https://github.com/wpilibsuite/roborio-toolchain/releases/download/v2020-2/FRC-2020-Linux-Toolchain-7.3.0.tar.gz";
+}
+
+#[cfg(target_os = "macos")]
+mod os {
+    use std::path::PathBuf;
+
+    pub fn install_dir() -> PathBuf {
+        dirs::home_dir()
+            .expect("Can't determine home directory")
+            .join("wpilib")
+        //.ok_or("Can't determine home directory".to_owned())
+    }
+
+    const TOOLCHAIN_URL_2019: &str = "https://github.com/wpilibsuite/toolchain-builder/releases/download/v2019-3/FRC-2019-Mac-Toolchain-6.3.0.tar.gz";
+    const TOOLCHAIN_URL_2020: &str = "https://github.com/wpilibsuite/roborio-toolchain/releases/download/v2020-2/FRC-2020-Mac-Toolchain-7.3.0.tar.gz";
+}
+
+//#[cfg(target_os = "windows")]
+//mod os {
+//    const TOOLCHAIN_URL_2019: &str = "https://github.com/wpilibsuite/toolchain-builder/releases/download/v2019-3/FRC-2019-Windows-Toolchain-6.3.0.zip";
+//    const TOOLCHAIN_URL_2020: &str = "https://github.com/wpilibsuite/roborio-toolchain/releases/download/v2020-2/FRC-2020-Windows-Toolchain-7.3.0.zip";
+//}
+
+#[cfg(unix)]
+pub fn install(toolchain: Toolchain) -> Result<(), String> {
+    if toolchain.installed() {
+        info!("Requested toolchain appears to already be installed");
+        return Err("Requested toolchain appears to already be installed".to_owned());
+    }
+
+    fs::create_dir_all(toolchain.path())
+        .map_err(str_map("Could not create toolchain install directory"))?;
+
+    if !Command::new("sh")
+        .arg("-c")
+        .arg(format!(
+            "wget -c {} -O - | tar -xz -C {} --strip-components=1",
```

Comment:
Yikes.

Reply:
I agree, but `tar` needs to be called no matter what so it seemed worthwhile to use `wget` to make downloading more convenient. The only thing I really don't like Is the amount of output from `wget`, but I just found a way to display a progress bar without everything else, so I'll change to that.

Reply:
Surely there's a `tar` crate?

Reply:
There is, but it does tar and nothing else, and requires some extra work to replicate `--strip-components=1`. Downloading and installing without using `sh` would require adding `reqwest` (which could also bring async into the equation), `flate2`, and `tar`. Since `cargo-frc` already heavily relies on calling unix commands instead of using libraries, I figured this would be a very convenient alternative. Do you think it's very important or advantageous to change to use these libraries?

Reply:
something something Windows